azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -96,23 +96,14 @@ public ExecutorLoader getExecutorLoader() {
    * contain an event for all running flows. This could happen, for example, during webserver
    * restarts.
    * @param event pod watch event
+   * @returns true if transition is valid else false
    */
-  private void validateTransition(AzPodStatusMetadata event) {
-    AzPodStatus currentStatus = getCurrentAzPodStatus(event.getPodName());
+  private boolean validateTransition(AzPodStatusMetadata event, AzPodStatus currentStatus) {
```

Comment:
nit: I thought we're put final before the method params, this applies to all methods in this class.

Reply:
There is an internal ticket for adding `final` for all the params in this (sub)package. I would prefer to defer this change for that ticket as this will also minimize unrelated changes showing up in the interim reviews.